### PR TITLE
Enable global kernel for legacy compatibility for Behat tests

### PR DESCRIPTION
**Pull request by matks**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Some core prestashop classes require a global kernel declared. In order to mimic this environment, I added it to Behat fixtures. I also added a fake employee login in order to allow some services (that require one) to work.
| Type?         | bug fix
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | 
| How to test?  | Travis is green

Should help https://github.com/PrestaShop/PrestaShop/pull/13554

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/13826)

### CODE REVIEW

**Comment by matks** [comment]
@sarjon @tomlev wdyt ?

Since there was no global `$kernel`, behat tests were skipping one step in StockManager:
https://github.com/PrestaShop/PrestaShop/blob/develop/src/Core/Stock/StockManager.php#L354

I fixed it by (😭) making the Behat SF kernel global, but I had to add a fake logged in Employee as StockManagementRepository requires one.

https://github.com/PrestaShop/PrestaShop/blob/develop/src/PrestaShopBundle/Entity/Repository/StockManagementRepository.php#L130

File: tests/Integration/Behaviour/Features/Context/CommonFeatureContext.php [hunk 1]
```diff
@@ -49,6 +49,12 @@ public static function prepare($scope)
 
         self::$kernel = new AppKernel('test', true);
         self::$kernel->boot();
+
+        global $kernel;
+        $kernel = self::$kernel;
+
+        $employeeId = \Employee::employeeExists('test@prestashop.com');
```
Comment (Quetzacoalt91) [inline tests/Integration/Behaviour/Features/Context/CommonFeatureContext.php]:
```php
/**
     * Check if Employee exists.
     *
     * @param string $email Employee email
     *
     * @return bool Indicates whether the Employee exists
     */
    public static function employeeExists($email)
    {
        if (!Validate::isEmail($email)) {
            die(Tools::displayError());
        }

        return (bool) Db::getInstance()->getValue('
		    SELECT `id_employee`
		    FROM `' . _DB_PREFIX_ . 'employee`
		    WHERE `email` = \'' . pSQL($email) . '\'
        ');
    }
```

:thinking:

File: tests/Integration/Behaviour/Features/Context/CommonFeatureContext.php [hunk 2]
```diff
@@ -49,6 +49,12 @@ public static function prepare($scope)
...
+        \Context::getContext()->employee = new \Employee($employeeId);
```
Comment (Quetzacoalt91) [inline tests/Integration/Behaviour/Features/Context/CommonFeatureContext.php]: I guess hardcoding 1 as paremeter would be enough for the tests, don't you think?

File: tests/Integration/Behaviour/Features/Context/CommonFeatureContext.php (hunk 1 above)
Comment (sarjon) [reply tests/Integration/Behaviour/Features/Context/CommonFeatureContext.php]: Docs are lying (once again).

Comment (PierreRambaud) [reply tests/Integration/Behaviour/Features/Context/CommonFeatureContext.php]: @sarjon Where is the lie? The result is casted. If you need the employee, use `getByEmail` method instead.

File: tests/Integration/Behaviour/Features/Context/CommonFeatureContext.php (hunk 2 above)
Comment (PierreRambaud) [inline tests/Integration/Behaviour/Features/Context/CommonFeatureContext.php]:
```suggestion
        $employee = new \Employee();
        \Context::getContext()->employee = $employee->getByEmail('test@prestashop.com');
```

File: tests/Integration/Behaviour/Features/Context/CommonFeatureContext.php (hunk 1 above)
Comment (sarjon) [reply tests/Integration/Behaviour/Features/Context/CommonFeatureContext.php]: Ohh, you are right.

File: tests/Integration/Behaviour/Features/Context/CommonFeatureContext.php (hunk 2 above)
Comment (matks) [reply tests/Integration/Behaviour/Features/Context/CommonFeatureContext.php]: Why not

File: tests/Integration/Behaviour/Features/Context/CommonFeatureContext.php [hunk 3]
```diff
@@ -49,6 +49,13 @@ public static function prepare($scope)
 
         self::$kernel = new AppKernel('test', true);
         self::$kernel->boot();
+
+        global $kernel;
+        $kernel = self::$kernel;
+
+        $employeeId = \Employee::employeeExists('test@prestashop.com');
```
Comment (Quetzacoalt91) [inline tests/Integration/Behaviour/Features/Context/CommonFeatureContext.php]:
```suggestion
```

**Review by Quetzacoalt91 — COMMENTED** [review]
An unused variable can be removed.

**Comment by sarjon** [comment]
Is there anything else left to do?

**Comment by jolelievre** [comment]
> Is there anything else left to do?

Removing the useless line with `$employeeId = \Employee::employeeExists('test@prestashop.com');
`

**Review by PierreRambaud — APPROVED** [review]

**Comment by matks** [comment]
Thank you for review

**Pull request merged**